<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1301</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,4">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="pushButton_settings">
            <property name="text">
             <string>配置参数</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_readData">
            <property name="text">
             <string>读取数据</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit">
            <property name="plainText">
             <string>1.读取las文件
2.选择去噪模式
3.设定图像显示的阈值(默认上下阈值分别为1，0，设置较小的上下阈值有利于查看结果)
4.点击去噪
5.鼠标右键点击图像并拖动，进行交互去噪，交互去噪的阈值和半径需要手动设置（半径可根据图像坐标尺寸来选择，阈值根据去噪效果来选择，当去噪效果不理想时点击上一步，即可恢复上一步的图像显示）
交互去噪有三种类型
a.鼠标右键单击去噪 
b.鼠标右键滑动去噪
c.鼠标左键选择矩形框去噪
6.去噪完毕后，可以导出数据或者导出图像（导出图像可指定图像的尺寸，一般默认选择宽500像素，高1000像素）
</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>20</height>
             </size>
            </property>
            <property name="text">
             <string>常规去噪模式</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_denoiseMode"/>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_denoisePrc">
            <property name="text">
             <string>去噪</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>上阈值</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_vmax"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>下阈值</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_vmin"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>交互去噪-阈值</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_threshold">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>20</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>交互去噪-半径</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_radius"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="pushButton_saveData">
            <property name="text">
             <string>导出数据</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_savePic">
            <property name="text">
             <string>导出图像</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="1,100">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0,0,0,0,0,0,0,0,0,0">
          <item>
           <widget class="QPushButton" name="pushButton_back">
            <property name="text">
             <string>上一步</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_oriScale">
            <property name="text">
             <string>恢复初始大小</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>交互去噪模式</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_InteractionMode"/>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_rightClick">
            <property name="text">
             <string>单击交互</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>去除噪声计算模式</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_calDenoiseVal"/>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_whichone">
            <property name="text">
             <string>中心or边缘</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_rect">
            <property name="text">
             <string>矩形框交互</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_rightSlider">
            <property name="text">
             <string>滑动交互</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="widget_pic" native="true"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1301</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
